src/scripts/cloud_error_processing.py

```diff
@@ -0,0 +1,80 @@
+import os
+import json
+import requests
+import sys
+from google.cloud import firestore
+from googleapiclient.discovery import build
+from google.cloud import secretmanager
+
+# Add our library utils to the path
+sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), "..")))
+
+from lib.constants import ISSUES_API_URL, GCS_PROJECT_NAME
+
+
+class GithubIssueHandler:
+    """ Handles posting issues to github using the personal access token stored in SecretsManager. """
+
+    def __init__(self):
+        self._username = "automation"  # This is ignored because we use an access token.
+        self._password = self._get_github_token()
+
+    def _error_group_to_github_issue(self, error_group):
+        title = "Automated error report"
+        body = error_group["representative"]["message"]
+        return {"title": title, "body": body}
+
+    def _get_github_token(self):
+        client = secretmanager.SecretManagerServiceClient()
+        name = client.secret_version_path(GCS_PROJECT_NAME, "github-token", "latest")
+        response = client.access_secret_version(name)
+        return response.payload.data
+
+    def post_error_to_github(self, error_group):
+        """ Returns the issue url if successfully posted, else raises a ConnectionError exception """
+        session = requests.Session()
+        session.auth = (self._username, self._password)
+        response = session.post(ISSUES_API_URL, json=self._error_group_to_github_issue(error_group))
+        if response.status_code != 201:
+            raise ConnectionError(
+                "Could not create github issue, bad status code received: {}".format(
+                    response.status_code
+                )
+            )
+        return response.json()["url"]
+
+
+def register_new_errors():
+    """ If new error groups are reported, log an issue on github with the details """
+    service = build("clouderrorreporting", "v1beta1")
+    errors_past_day = (
+        service.projects()
+        .groupStats()
+        .list(projectName="projects/{}".format(GCS_PROJECT_NAME), timeRange_period="PERIOD_1_DAY")
+        .execute()
+    )
+    gh_issue_handler = GithubIssueHandler()
+    db = firestore.Client()
+    errors_db = db.collection("errors")
+    for error_group in errors_past_day["errorGroupStats"]:
+        group_id = error_group["group"]["groupId"]
+        doc = errors_db.document(group_id)
+        if not doc.get().exists and error_group["group"]["resolutionStatus"] == "OPEN":
+            try:
+                error_group["group"]["trackingIssues"] = gh_issue_handler.post_error_to_github(
+                    error_group
+                )
+                error_group["group"]["resolutionStatus"] = "ACKNOWLEDGED"
+            except:
+                # Could not create an issue
+                # Don't add it to our known errors db, we can retry on the next scheduled job.
+                continue
```

Comment:
Maybe log the error using the class `ErrorLogger`?

Reply:
The original class now logs an exception. I've also changed it so that this try only catches ConnectionError, other errors will remain uncaught.